code/modules/mob/mob.dm

```diff
@@ -52,6 +53,8 @@
 		move_intent = move_intents[1]
 	if(ispath(move_intent))
 		move_intent = decls_repository.get_decl(move_intent)
+	if(ispath(ai))
+		ai = new(src)
```

Comment:
Even if you do not use the proc override this should instead be `ai = new ai(src)`, otherwise you'll only instantiate the type of the var `ai`, in this case `/datum/ai`, not whatever path that is assigned to it.